src/kOS/UserIO/TelnetSingletonServer.cs

```diff
@@ -170,15 +170,26 @@ public TelnetSingletonServer(TelnetMainServer mainServer, TcpClient client, int
             rawStream = client.GetStream();
             lock (keepAliveAccess)
                 gotSomeRecentTraffic = true;
-            inThread = new Thread(DoInThread);
-            outThread = new Thread(DoOutThread);
             outQueue = new Queue<char>();
             inQueue = new Queue<char>();
+            SpawnWelcomeMenu();
+            inThread = new Thread(DoInThread);
+            outThread = new Thread(DoOutThread);
             ClientWidth = 80; // common default for a lot of terminal progs.  Will allow resize via RFC1073.
             ClientHeight = 24; // common default for a lot of terminal progs.  Will allow resize via RFC1073.
             ClientTerminalType = "INITIAL_UNSET"; // will be set by telnet client as described by RFC 1091
         }
 
+        ~TelnetSingletonServer()
+        {
+            if (welcomeMenu != null)
+            {
+                welcomeMenu.Detach();
+                GameObject.Destroy(welcomeMenu);
+                welcomeMenu = null;
+            }
+        }
```

Comment:
Since we no longer get rid of ``welcomeMenu`` and re-make a new one each time, the final point where we really do get rid of it has to wait until we're all done with this telnet singleton client.